tests/ballerina-test/src/test/java/org/ballerinalang/test/services/nativeimpl/inbound/request/LocalAddressNativeFunctionTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.services.nativeimpl.inbound.request;
+
+import org.ballerinalang.launcher.util.BServiceUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BJSON;
+import org.ballerinalang.net.http.HttpConstants;
+import org.ballerinalang.test.services.testutils.HTTPTestRequest;
+import org.ballerinalang.test.services.testutils.MessageUtils;
+import org.ballerinalang.test.services.testutils.Services;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.transport.http.netty.message.HTTPCarbonMessage;
+import org.wso2.transport.http.netty.message.HttpMessageDataStreamer;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Test cases for ballerina.net.http.getLocalAddress native functions.
+ */
+public class LocalAddressNativeFunctionTest {
+
+    private CompileResult serviceResult;
+
+    @BeforeClass
+    public void setup() {
+        String filePath = "test-src/statements/services/nativeimpl/inbound/request/server-address-native-function.bal";
+        serviceResult = BServiceUtil.setupProgramFile(this, filePath);
+    }
+
+    @Test(description = "Test whether getLocalAddress native function populates the correct value.")
+    public void testGetRemoteAddress() {
```

Comment:
This tests can be moved to the InRequestNativeFunctionSuccessTest class I suppose. All the native functions bound to InRequest are located there.